src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -76,32 +77,18 @@ void ElementView::FUNC(){ \
    Gui::Application::Instance->commandManager().runCommandByName(CMDSTR);}
 
 // helper class to store additional information about the listWidget entry.
-class ElementItem : public QListWidgetItem
+class ElementData
 {
 public:
-    ElementItem(const QIcon & icon, const QString & text, int elementnr,
-                int startingVertex, int midVertex, int endVertex,
-                Base::Type geometryType, bool construction, bool external)
-        : QListWidgetItem(icon,text)
-        , ElementNbr(elementnr)
-        , StartingVertex(startingVertex)
-        , MidVertex(midVertex)
-        , EndVertex(endVertex)
-        , isLineSelected(false)
-        , isStartingPointSelected(false)
-        , isEndPointSelected(false)
-        , isMidPointSelected(false)
-        , GeometryType(geometryType)
-        , isConstruction(construction)
-        , isExternal(external)
-    {
 
-    }
-    ElementItem(const QString & text,int elementnr,
-                int startingVertex, int midVertex, int endVertex,
-                Base::Type geometryType, bool construction, bool external)
-        : QListWidgetItem(text)
-        , ElementNbr(elementnr)
+    ElementData() = default;
+    ~ElementData() = default;
+    ElementData(const ElementData&) = default;
+    ElementData& operator=(const ElementData&) = default;
+
+    ElementData(int elementnr, int startingVertex, int midVertex, int endVertex,
+        Base::Type geometryType, bool construction, bool external, QIcon ic0, QIcon ic1, QIcon ic2, QIcon ic3, QString lab) :

```

Comment:
This is heavy. "pass by value" icons and strings should be very exceptional. Copies are being created on every object construction. This is why in the old ElementItem class, they were passed by reference. With reference to the icons, I really wonder whether this is not better handled by using the "geometryType" as key for a factory (see comment in the .h file), so that it is easy to get pointers to pre-built icons (built only once). This needs reconsideration.